java/src/jmri/jmrix/dcc4pc/Dcc4PcReporter.java

```diff
@@ -23,26 +25,17 @@ public Dcc4PcReporter(String systemName, String userName) {  // a human-readable
         super(systemName, userName);  // can't use prefix here, as still in construction
     }
 
-    @Override
-    public void dispose() {
-        super.dispose();
-    }
-
     // data members
     transient RailComPacket[] rcPacket = new RailComPacket[3];
 
-    void setPacket(int[] arraytemp, int dcc_addr_type, int dcc_addr, int cvNumber, int speed, int packetTypeCmd) {
-        if (log.isDebugEnabled()) {
-            log.debug("{} dcc_addr {} {} {}", getDisplayName(), dcc_addr, cvNumber, speed);
-        }
+    synchronized void setPacket(int[] arraytemp, int dcc_addr_type, int dcc_addr, int cvNumber, int speed, int packetTypeCmd) {
+        log.debug("{} dcc_addr {} {} {}", getDisplayName(), dcc_addr, cvNumber, speed);
         RailComPacket rc = new RailComPacket(arraytemp, dcc_addr_type, dcc_addr, cvNumber, speed);
         decodeRailComInfo(rc, packetTypeCmd);
```

Comment:
Why is the method `setPacket()` synchronized? `decodeRailComInfo()` is synchronized, but not on the instance of the object but on the `lock` variable. What is it you are trying to protect here?